### PR TITLE
Certificate mapping test

### PR DESCRIPTION
**Pull request by davidkupka**
https://fedorahosted.org/freeipa/ticket/6542

### CODE REVIEW

**Comment by martbab** [comment]
I have some inline comments. I was also thinking about the nomenclature of the Tracker mixins and I think we should name them based on the noun of the action that is being tracked, e.g:

RetrievalTracker
SearchTracker
CreationTracker
ModificationTracker

File: ipatests/test_xmlrpc/tracker/certmap_plugin.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+from ipatests.util import assert_deepequal
+
+
+class CertmapruleTracker(Tracker, EnableTracker):
+    """ Tracker for testin certmaprule plugin """
+    retrieve_keys = {
+        u'dn',
+        u'cn',
+        u'description',
+        u'ipacertmapissuer',
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/tracker/certmap_plugin.py]: you can remove ipacertmapissuer attribute

File: ipatests/test_xmlrpc/tracker/certmap_plugin.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                 default_version=None):
+        super(CertmapruleTracker, self).__init__(
+            default_version=default_version)
+
+        self.dn = DN((u'cn', cn,),
+                     self.api.env.container_certmaprules,
+                     self.api.env.basedn)
+        self.options = {
+            u'description': description,
+            u'ipacertmapissuer': ipacertmapissuer,
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/tracker/certmap_plugin.py]: you can remove ipacertmapissuer attribute.

File: ipatests/test_xmlrpc/tracker/certmap_plugin.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        u'dn',
+        u'ipacertmappromptusername',
+    }
+
+    retrieve_all_keys = retrieve_keys | {
+        u'cn',
+        u'objectclass',
+        u'aci',
+        u'ipacertmapversion',
+        u'ipaconfigstring'
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/tracker/certmap_plugin.py]: you can remove ipacertmapversion and ipaconfigstring

File: ipatests/test_xmlrpc/tracker/certmap_plugin.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    def __init__(self, default_version=None):
+        super(CertmapconfigTracker, self).__init__(
+            default_version=default_version)
+
+        self.attrs = {
+            u'dn': DN(self.api.env.container_certmap, self.api.env.basedn),
+            u'cn': [self.api.env.container_certmap[0].value],
+            u'objectclass': objectclasses.certmapconfig,
+            u'ipacertmapversion': [u'1'],
+            u'ipaconfigstring': [u'CertMapVersion 1'],
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/tracker/certmap_plugin.py]: you can remove ipacertmapversion and ipaconfigstring

File: ipatests/test_xmlrpc/tracker/certmapdata.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        if not any_in(self.certmapdata_options, options):
+            raise RequirementError(name=u'certmapdata')
+
+        if (
+            any_in((u'ipacertmapdata', u'usercertificate',), options) and
+            any_in((u'issuer', u'subject',), options)
+        ):
+            raise MutuallyExclusiveError(reason=u'Mutually exclusive options '
+                                                u'provided at the same time.')
+
```
Comment (flo-renaud) [inline ipatests/test_xmlrpc/tracker/certmapdata.py]: You can also add a requirement: if subject or issuer is provided, then the other one is also required.

**Comment by stlaz** [comment]
This's been hanging here for long enough, why isn't it finished? I don't see any postponed label here, do we know whether it's somehow blocked?

**Pull request closed without merging**